src/main/java/com/example/analytics/service/coinMarketCap/CryptoPriceService.java

```diff
@@ -0,0 +1,55 @@
+package com.example.analytics.service.coinMarketCap;
+
+import com.example.analytics.exception.coinMarketCap.ApiException;
+import com.example.analytics.model.coinMarketCap.CryptoCurrencyData;
+import com.example.analytics.repository.coinMarketCap.CryptoCurrencyDataRepository;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+
+import static com.example.analytics.utils.PropertiesReader.getProperty;
+
+@Service
+public class CryptoPriceService {
+    private final String apiUrl = getProperty("crypto.base.url");
+    private final String apiKey = getProperty("crypto.api.key");
+    private final RestTemplate restTemplate;
+    private final CryptoCurrencyDataRepository cryptoCurrencyDataRepository;
+
+    public CryptoPriceService(RestTemplate restTemplate, CryptoCurrencyDataRepository cryptoCurrencyDataRepository) {
+        this.restTemplate = restTemplate;
+        this.cryptoCurrencyDataRepository = cryptoCurrencyDataRepository;
+    }
+
+    public String fetchCryptoPrice(String symbol, String convert) {
```

Comment:
Could you return the already predefined model CryptoCurrencyData instead of String?

Reply:
fixed, I have already implemented it